lib/scholar/naive_bayes/gaussian_naive_bayes.ex

```diff
@@ -0,0 +1,442 @@
+            "wrong input rank. Expected x to be rank 2 and y to be rank 1, got: #{input_rank} for x and #{targets_rank} for y"
+    end
+
+    {num_samples, _} = Nx.shape(x)
+    {num_targets} = Nx.shape(y)
+
+    if num_samples != num_targets do
+      raise ArgumentError,
+            "wrong input shape. Expected x to have the same first dimension as y, got: #{num_samples} for x and #{num_targets} for y"
+    end
+
+    opts =
+      [
+        sample_weights_flag: if(opts[:sample_weights] != nil, do: true, else: false),
+        priors_flag: if(opts[:priors] != nil, do: true, else: false)
+      ] ++
+        opts
+
+    {sample_weights, opts} = Keyword.pop(opts, :sample_weights, 1.0)
+    sample_weights = Nx.tensor(sample_weights)
+
+    {priors, opts} = Keyword.pop(opts, :priors, 0.0)
+    class_priors = Nx.tensor(priors)
+
+    fit_n(x, y, sample_weights, class_priors, opts)
+  end
+
+  @doc """
+  Perform classification on an array of test vectors `x` using `model`.
+
+  ## Examples
+
+      iex> x = Nx.iota({4, 3})
+      iex> y = Nx.tensor([1, 2, 0, 2])
+      iex> model = Scholar.NaiveBayes.Gaussian.fit(x, y, num_classes: 3)
+      iex> Scholar.NaiveBayes.Gaussian.predict(model, Nx.tensor([[6, 2, 4], [8, 5, 9]]))
+      #Nx.Tensor<
+        s64[2]
+        [2, 2]
+      >
+  """
+  defn predict(%__MODULE__{classes: classes} = model, x) do
+    check_input(model, x)
+    jll = joint_log_likelihood(model, x)
+    Nx.take(classes, Nx.argmax(jll, axis: 1))
+  end
+
+  @doc """
+  Return joint log probability estimates for the test vector `x` using `model`.
+
+  ## Examples
+
+      iex> x = Nx.iota({4, 3})
+      iex> y = Nx.tensor([1, 2, 0, 2])
+      iex> model = Scholar.NaiveBayes.Gaussian.fit(x, y, num_classes: 3)
+      iex> Scholar.NaiveBayes.Gaussian.predict_joint_log_probability(model, Nx.tensor([[6, 2, 4], [8, 5, 9]]))
+      #Nx.Tensor<
+        f32[2][3]
+        [
+          [-1822222336.0, -1822222208.0, -9.023576736450195],
+          [-399999968.0, -5733332992.0, -7.245799541473389]
+        ]
+      >
+  """
+  defn predict_joint_log_probability(%__MODULE__{} = model, x) do
+    check_input(model, x)
+    joint_log_likelihood(model, x)
+  end
+
+  @doc """
+  Return log-probability estimates for the test vector `x` using `model`.
+
+  ## Examples
+
+      iex> x = Nx.iota({4, 3})
+      iex> y = Nx.tensor([1, 2, 0, 2])
+      iex> model = Scholar.NaiveBayes.Gaussian.fit(x, y, num_classes: 3)
+      iex> Scholar.NaiveBayes.Gaussian.predict_log_probability(model, Nx.tensor([[6, 2, 4], [8, 5, 9]]))
+      #Nx.Tensor<
+        f32[2][3]
+        [
+          [-1822222336.0, -1822222208.0, 0.0],
+          [-399999968.0, -5733332992.0, 0.0]
+        ]
+      >
+  """
+  defn predict_log_probability(%__MODULE__{} = model, x) do
+    check_input(model, x)
+    jll = joint_log_likelihood(model, x)
+
+    log_proba_x =
+      jll
+      |> Nx.exp()
+      |> Nx.sum(axes: [1])
+      |> Nx.log()
+      |> Nx.reshape({:auto, 1})
```

Comment:
```suggestion
      |> Nx.new_axis(1)
```
assuming this is already a flat tensor

Reply:
Same result and efficiency, I just think it's clearer on what's happening